src/triangle.rs

```diff
@@ -174,17 +189,21 @@ impl Triangle {
     // TODO: maybe make height functions for right and non right angled triangles
 
     /// The area of a right angled [`Triangle`]
+    ///
+    /// NOTE: assumed [`Triangle::organise`]d positions
     #[inline]
     pub fn right_angled_area(&self) -> f64 {
         // A = 1/2*b*h
-        (self.b.distance(self.c) * self.b.midpoint(self.c).distance(self.a)) / 2.0
+        (self.b.distance(self.c) * (self.a.y - self.b.y)) / 2.0
     }
 
     /// Updates [`Triangle`]'s vertice positions
     /// such that [`Triangle::a`] is topmost, [`Triangle::b`] is leftmost, and [`Triangle::c`] is rightmost
     #[inline]
     pub fn organise(self) -> Self {
-        Self::new(self.topmost(), self.leftmost(), self.rightmost())
+        let t = Self::new(self.topmost(), self.leftmost(), self.rightmost());
+
+        t
     }
```

Comment:
(ignoring the bug which will be fixed, im just trying to find a clean way) there are a few issues / design choices that can be made here:
- should organise() mutate `self`?
- a lot of other functions assume an organised position so should we either let the user call organised before these functions OR call organise in these functions